adapter/src/test/java/com/twilio/kudu/sql/PaginationIT.java

```diff
@@ -673,4 +705,55 @@ public void testPaginationMultipleAccountsOrderedByAccountSid() throws Exception
     }
   }
 
+  @Test
+  public void testPaginationMultipleAccountsOrderedByAccountSidUsingINWithHint() throws Exception {
+    try (Connection conn = DriverManager.getConnection(JDBC_URL)) {
+      TimestampString lowerBoundDateInitiated = TimestampString.fromMillisSinceEpoch(T1);
+      TimestampString upperBoundDateInitiated = TimestampString.fromMillisSinceEpoch(T4);
+      String dateInitiatedOrder = descending ? "DESC" : "ASC";
+
+      String firstBatchSqlFormat = "SELECT * FROM %s %s WHERE account_sid IN ('ACCOUNT1','ACCOUNT2') AND date_initiated >= TIMESTAMP'%s' AND date_initiated < TIMESTAMP'%s' "
+          + "ORDER BY account_sid, date_initiated %s, transaction_id " + "LIMIT 7";
+
+      String firstBatchSql = String.format(firstBatchSqlFormat, tableName, hint, lowerBoundDateInitiated,
+          upperBoundDateInitiated, dateInitiatedOrder);
+      ResultSet rs = conn.createStatement().executeQuery("EXPLAIN PLAN FOR " + firstBatchSql);
+      String plan = SqlUtil.getExplainPlan(rs);
+
+      String expectedPlanFormat = "KuduToEnumerableRel\n"
+          + "  KuduSortRel(sort0=[$0], sort1=[$1], sort2=[$2], dir0=[ASC], dir1=[%s], dir2=[ASC], "
+          + "fetch=[7], groupBySorted=[false])\n"
+          + "    KuduFilterRel(ScanToken 1=[account_sid EQUAL %s, date_initiated GREATER_EQUAL %d, "
+          + "date_initiated LESS %d], ScanToken 2=[account_sid EQUAL %s, date_initiated "
+          + "GREATER_EQUAL %d, date_initiated LESS %d])\n" + "      KuduQuery(table=[[kudu, %s]])\n";
+      String expectedPlan = String.format(expectedPlanFormat, dateInitiatedOrder, ACCOUNT1, T1 * 1000, T4 * 1000,
+          ACCOUNT2, T1 * 1000, T4 * 1000, tableName);
+      assertEquals(String.format("Unexpected plan\n%s", plan), expectedPlan, plan);
+    }
+  }
+
+  @Test
+  public void testPaginationMultipleAccountsOrderedByAccountSidUsingINWithoutHint() throws Exception {
```

Comment:
Trying to paginate when using an in list of the leading pk column by without using the hint will lead to rows returning incorrect results if the `KuduSortRel` is used. 
Is there a way in `KuduFilterVisitor` to check if this hint is set in the OR case and if if its not just return false so that `KuduSortRel` will not be matched in that case?

Reply:
I was hoping to use while paginating as well.

Reply:
made the suggested changes and added a test

Reply:
nit: I think removing the  `UsingIN` in the test would make it cleaner.